pkg/utils/utils.go

```diff
@@ -12,3 +15,23 @@ func CalculateDuration(start, end time.Time) string {
 func FormatDate(t time.Time) string {
 	return t.Format(DateLayoutFormat)
 }
+
+// Common function to calculate test metrics
+func CalculateTestMetrics(testRuns []models.TestRun) (totalTests, executedTests, passedTests, failedTests int) {
+	for _, testRun := range testRuns {
+		for _, suiteRun := range testRun.SuiteRuns {
+			for _, specRun := range suiteRun.SpecRuns {
+				totalTests++ // Count each spec run
+				if specRun.Status != "skipped" {
```

Comment:
nit: use constants for status values

const (
	StatusSkipped = "skipped"
	StatusPassed  = "passed"
	StatusFailed  = "failed"
)

Reply:
Updated to use constants and squashed the commits to include DCO